spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/domain/EntityScanner.java

```diff
@@ -63,7 +78,19 @@ public EntityScanner(ApplicationContext context) {
 		if (packages.isEmpty()) {
 			return Collections.emptySet();
 		}
-		ClassPathScanningCandidateComponentProvider scanner = new ClassPathScanningCandidateComponentProvider(false);
+		ClassPathScanningCandidateComponentProvider scanner;
+		if (this.excludeAbstractClasses) {
```

Comment:
Rather than doing all of this, I would expose a `protected` method where the `ClassPathScanningCandidateComponentProvider` can be tuned. You'd then be able to change its behaviour without having to duplicate the logic.

Reply:
That makes sense. I 'll make those changes the week of 17 of August. Thanks for the review!